### ISSUE TITLE
[mechanics] build fails on ubuntu trusty (14.04) (bullet 2.81)

### ISSUE THREAD

**Issue by xhub** [siconos/siconos#65]
/«PKGBUILDDIR»/mechanics/src/collision/bullet/BulletBroadphase.cpp: In member function ‘virtual void BulletBroadphase::visit(SP::SiconosConvexHull)’:
/«PKGBUILDDIR»/mechanics/src/collision/bullet/BulletBroadphase.cpp:568:32: error: no matching function for call to ‘btConvexHullShape::addPoint(const btVector3&, bool)’
         btch->addPoint(v, false);
                                ^
/«PKGBUILDDIR»/mechanics/src/collision/bullet/BulletBroadphase.cpp:568:32: note: candidate is:
In file included from /«PKGBUILDDIR»/mechanics/src/collision/bullet/BulletBroadphase.cpp:58:0:
/usr/include/bullet/BulletCollision/CollisionShapes/btConvexHullShape.h:39:7: note: void btConvexHullShape::addPoint(const btVector3&)
  void addPoint(const btVector3& point);

Full log:

https://launchpadlibrarian.net/264847417/buildlog_ubuntu-trusty-amd64.siconos_3.9~alpha-0~5236~ubuntu14.04.1_BUILDING.txt.gz

**Comment by vacary** [comment]
The same on MacOsC El Capitan with clan G

> Apple LLVM version 7.3.0 (clang-703.0.31)
> Target: x86_64-apple-darwin15.5.0
> Thread model: posix
> InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
>

**Comment by radarsat1** [comment]
Reverted in 3f0ab095ab

**Comment by xhub** [comment]
Still there (see https://launchpadlibrarian.net/301226707/buildlog_ubuntu-trusty-amd64.siconos_3.9~alpha-0~5689~ubuntu14.04.1_BUILDING.txt.gz )

/<<PKGBUILDDIR>>/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp: In member function ‘void SiconosBulletCollisionManager_impl::createCollisionObject(SP::SiconosVector, SP::BodyDS, SP::SiconosConvexHull, SP::SiconosContactor)’:
/<<PKGBUILDDIR>>/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:827:32: error: no matching function for call to ‘btConvexHullShape::addPoint(const btVector3&, bool)’
         btch->addPoint(v, false);
                                ^
/<<PKGBUILDDIR>>/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:827:32: note: candidate is:
In file included from /<<PKGBUILDDIR>>/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:70:0:
/usr/include/bullet/BulletCollision/CollisionShapes/btConvexHullShape.h:39:7: note: void btConvexHullShape::addPoint(const btVector3&)
  void addPoint(const btVector3& point);
       ^
/usr/include/bullet/BulletCollision/CollisionShapes/btConvexHullShape.h:39:7: note:   candidate expects 1 argument, 2 provided
make[3]: *** [mechanics/CMakeFiles/mechanics.dir/src/collision/bullet/SiconosBulletCollisionManager.cpp.o] Error 1


We can also decide not to support trusty.

**Comment by radarsat1** [comment]
Should be fixed in 10144e1d, but I still get build errors under a Trusty container, like this:
~~~
[ 88%] Swig source
Unable to open file /root/build/control/swig/control/controllerPYTHON_wrap.cxx: No such file or directory
~~~
It seems swig is running succesfully but then the output is not found.  Since I had to modify `CMakeLists.txt` to change to an older version of CMake, let's see if this isn't just a build system error.. if so, and launchpad fails, let's drop Trusty support due to an out of date CMake.

**Comment by xhub** [comment]
Fixed indeed, thanks
